docs/style-guide.rst

```diff
@@ -867,7 +867,70 @@ indistinguishable from the numerals one and zero.
 Contract and Library Names
 ==========================
 
-Contracts and libraries should be named using the CapWords style. Examples: ``SimpleToken``, ``SmartBank``, ``CertificateHashRepository``, ``Player``.
+* Contracts and libraries should be named using the CapWords style. Examples: ``SimpleToken``, ``SmartBank``, ``CertificateHashRepository``, ``Player``, ``Congress``, ``Owned``. 
+* Contract and Library names must match their Constructor name (otherwise it is just compiled as a Function rather than as a Constructor, since a Constructor is a Function with the same name as the associated Contract or Library name that it belongs to and may be optionally included in the Contract or Library). 
+* Contract and Library names should also match their Filenames. 
+* If a Contract File includes a Core Contract in addition to other Contracts and Libraries that are all contained within the same file then the Core Contract name should match the Solidity Contract Filename. Refer to the Contract Source Code of CryptoKitties https://etherscan.io/address/0x06012c8cf97bead5deae237070f9587f8e7a266d#code
```

Comment:
I don't think we should refer to CryptoKitties / etherscan here. Especially as etherscan requires all sourcecode to be concatenated for verification.